puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/tarballsRelease.groovy

```diff
@@ -0,0 +1,106 @@
+pipeline {
+    agent { label 'admin && sshkey' }
+
+    options {
+        timestamps()
+        timeout(time: 2, unit: 'HOURS')
+        disableConcurrentBuilds()
+        ansiColor('xterm')
+    }
+
+    def version = env.getProperty('version')
+    def major_version = env.getProperty('major_version')
+    def ruby_ver = '2.4.0'
+
+    stages {
+        stage('Parallel') {
+
+            parallel {
+                stage('Foreman') {
+                    steps {
+                        verify_tag('foreman', version)
+                        build_tarball('foreman', version, ruby_ver)
+                    }
+                }
+                stage('Smart Proxy') {
+                    steps {
+                        verify_tag('smart-proxy', version)
+                        build_tarball('smart-proxy', version, ruby_ver)
+                    }
+                }
+                stage('Foreman SELinux') {
+                    steps {
+                        verify_tag('foreman-selinux', version)
+                        build_tarball('foreman-selinux', version, ruby_ver)
+                    }
+                }
+                stage('Foreman Installer') {
+                    steps {
+                        verify_tag('foreman-installer', version)
+                        build_tarball('foreman-installer', version, ruby_ver)
+                    }
+                }
+            }
+        }
+    }
+
+    post {
+        success {
+            build job: 'release_packages', parameters: [
+                [$class: 'StringParameterValue', name: 'version', value: version],
+                [$class: 'StringParameterValue', name: 'major_version', value: major_version]
+            ]
+        }
+    }
+}
+
+void verify_tag(project, version) {
+    deleteDir()
+    dir(project) {
+        git url: "https://github.com/theforeman/${project}", branch: 'develop'
+        sh """
+        if [ \"$(git tag -l ${version} | wc -l)\" -ne 1 ]; then
+            echo \"ERROR: cannot find tag ${version}\"
+            git tag -l
+            exit 1
+        fi
+        exit 0
+        """
+    }
+    deleteDir()
+}
+
+void build_tarball(project, version, ruby_ver) {
+    def base_dir = "/var/www/vhosts/downloads/htdocs/foreman"
+    def rake = "rake"
+
+    deleteDir()
+    dir(project) {
+        git url: 'https://github.com/theforeman/foreman', branch: version
```

Comment:
Branches typically ```VERSION-stable```

Reply:
I believe `-stable` would be included in the `version` variable. It's not appended to the [current job](ci.theforeman.org/job/release_build_tarball) as seen below:

```sh
#!/bin/bash -xe

git checkout ${version}

if [ $project = foreman ]; then
  cat config/settings.yaml.example > config/settings.yaml
  cat config/database.yml.example > config/database.yml
fi

...
```

Reply:
But the exact same variable is used to verify the tag exists. That doesn't sound correct.

Reply:
`release_verify_tag` doesn't append `-stable` either.

```sh
#!/bin/bash -xe

if [ "$(git tag -l ${version} | wc -l)" -ne 1 ]; then
  echo "ERROR: cannot find tag ${version}"
  git tag -l
  exit 1
fi

exit 0
```

Reply:
I meant that both use ```$version```. If I understand it correct then verify_tag would check ```git tag -l ${version}``` (which is either 1.17.0 I guess). Then package building results in ```git checkout 1.17.0``` (which works). That does refer to a tag rather than a branch and I'm not sure ```branch: SOME_TAG``` works because I don't know how Jenkins internally works.

Reply:
wow that's surprising. the `git:` plugin actually **doesn't** support referencing by tag ([ref](https://issues.jenkins-ci.org/browse/JENKINS-27018)).